pkg/rancher-desktop/backend/wsl.ts

```diff
@@ -1392,169 +1392,168 @@ export default class WSLBackend extends events.EventEmitter implements VMBackend
-              // Do not await on this, as we don't want to wait until the proxy exits.
-              this.runWslProxy().catch(console.error);
-            }
-          }),
-          this.progressTracker.action('Installing image scanner', 100, this.installTrivy()),
-          this.progressTracker.action('Installing CA certificates', 100, this.installCACerts()),
-          this.progressTracker.action('Installing helpers', 50, this.installWSLHelpers()),
-        ];
+                    await this.writeFile(allowedImagesConf, patterns, 0o644);
+                  } else {
+                    await this.execCommand({ root: true }, 'rm', '-f', allowedImagesConf);
+                  }
+                  const obsoleteImageAllowListConf = path.join(path.dirname(allowedImagesConf), 'image-allow-list.conf');
 
-        if (kubernetesVersion) {
-          const version = kubernetesVersion;
-
-          installerActions.push(
-            this.progressTracker.action('Writing K3s configuration', 50, async() => {
-              const k3sConf = {
-                PORT:                   config.kubernetes.port.toString(),
-                LOG_DIR:                await this.wslify(paths.logs),
-                'export IPTABLES_MODE': 'legacy',
-                ENGINE:                 config.containerEngine.name,
-                ADDITIONAL_ARGS:        config.kubernetes.options.traefik ? '' : '--disable traefik',
-                USE_CRI_DOCKERD:        BackendHelper.requiresCRIDockerd(config.containerEngine.name, version).toString(),
-              };
-
-              // Make sure the apiserver can be accessed from WSL through the internal gateway
-              k3sConf.ADDITIONAL_ARGS += ' --tls-san gateway.rancher-desktop.internal';
+                  await this.execCommand({ root: true }, 'rm', '-f', obsoleteImageAllowListConf);
+                }),
+                await this.progressTracker.action('Rancher Desktop guest agent', 50, this.installGuestAgent(kubernetesVersion, this.cfg)),
+              ]);
 
+              await this.writeFile('/usr/local/bin/wsl-exec', WSL_EXEC, 0o755);
+              await this.runInit();
               if (rdNetworking) {
-                // Add the `veth-rd1` IP address from inside the namespace
-                k3sConf.ADDITIONAL_ARGS += ' --tls-san 192.168.1.2';
+              // Do not await on this, as we don't want to wait until the proxy exits.
+                this.runWslProxy().catch(console.error);
               }
+            }),
+            this.progressTracker.action('Installing image scanner', 100, this.installTrivy()),
+            this.progressTracker.action('Installing CA certificates', 100, this.installCACerts()),
+            this.progressTracker.action('Installing helpers', 50, this.installWSLHelpers()),
+          ]));
+
+          if (kubernetesVersion) {
+            const version = kubernetesVersion;
+
+            await this.progressTracker.action('Installing Kubernetes', 0, Promise.all([
```

Comment:
I wonder if we should add a comment here explaining that `kubeBackend.install` requires the container engine setup to be complete, so can't run in parallel?